include/imgproc/dct_hash.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+// Standard C++ includes
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// Standard C includes
+#include <cmath>
+
+//opencv
+#include <opencv2/opencv.hpp>
+
+
+class DCTHash {
+
+    private:
+    static unsigned int countSetBits(unsigned long long int n) { 
+        unsigned int count = 0; 
+        while (n) { 
+            count += n & 1; 
+            n >>= 1; 
+        } 
+        return count; 
```

Comment:
Good idea. I suppose an alternative would be to use ``std::bitset<64>`` instead of ``uint64_t`` for the hashes, then you've got the ``.count()`` method etc.

Reply:
That is an even better idea - much more C++y, _probably_ implements the optimisations I suggest and, if you do want to extract a ``uint64_t``, you can call ``to_ullong()``